example/pokeapi_functional/lib/api/fetch_pokemon.dart

```diff
@@ -52,3 +52,8 @@ TaskEither<String, Pokemon> fetchPokemon(int pokemonId) => TaskEither.tryCatch(
 /// All the functions are simply chained together following the principle of composability.
 TaskEither<String, Pokemon> fetchPokemonFromUserInput(String pokemonId) =>
     _validateUserPokemonId(pokemonId).flatMapTask(fetchPokemon);
+
+TaskEither<String, Pokemon> fetchRandomPokemon() => fetchPokemon(randomInt(
+      Constants.minimumPokemonId,
+      Constants.maximumPokemonId + 1,
+    ).run());
```

Comment:
You shouldn’t call `run()`. To keep the function pure, you should instead convert `IO` (returned by `randomInt`) to `TaskEither`.

Reply:
I am unfamiliar with FP and fpdart , can you tell me which method I should use?

Reply:
The function expects `TaskEither` as return type. `TaskEither` is an `async` operation (`Task`) that may fail (`Either`). 

`randomInt` instead returns `IO`. `IO` is a sync operation that cannot fail.

`fetchRandomPokemon` needs to execute the following operations one after the other (`pipe`):
1. Get random `int`
2. Use this `int` as a parameter for `fetchPokemon`

This means that we first need to execute `randomInt`, then `fetchPokemon`. To execute functions one after the other we can use `flatMap` in `fpdart`.

The problem is that `flatMap` on `IO` requires to return another `IO`. Therefore, we need to convert `IO` to `TaskEither`, and then call `flatMap`:
1. Call `randomInt` (returns `IO`)
2. Convert `IO` to `TaskEither`
3. Use `flatMap` to execute `fetchPokemon`

Does this help? Let me know if something is unclear, full support from me to anyone who contributes and asks for help 💯

> **Note**: I noticed that `IO` is missing a method `toTaskEither` (`IO` to `TaskEither` directly). Adding this to the list of todo for next release 🤝

Reply:
I added `toTaskEither` to `IO`. Now you can use `toTaskEither()` from `randomInt` instead of `run()`:
1. `randomInt`
2. `toTaskEither()`
3. `flatMap()`
4. `fetchPokemon`